modules/extreme_year.py

```diff
@@ -0,0 +1,117 @@
+"""
+this module will return the highest and lowest temperatures of a
+given year along with the most humid day.
+"""
+import glob
+
+
+def read_and_calculate_extremes(pattern, path):
+    """
+    this function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. max highest temperature
+    2. max lowest temperature
+    3. max humidity
+    :param pattern: a string which contains a year e.g '2006', '2007'
+    :param path: a string which contains path to a directory containing weather files
+    :return: a 3x2 list maximums where
+    maximums[0] = [max_highest_temperature, date]
+    maximums[1] = [max_lowest_temperature, date]
+    maximums[2] = [max_humidity, date]
+    date here is a string like: '2002-4-1', '2004-3-14'
+    """
+    files = glob.glob(path + f"*{pattern}*")
```

Comment:
Convert into a generic method, it can be reused.

Reply:
i didnt understand the comment. kindly elaborate

Reply:
I mean you might need to get the wildcard matching files in all the tasks. Writing a function in a file called `utils.py`, a function that takes `pattern` as a param and returns the matching files. In this way, you'd have a generic function that you can maybe use in all the tasks.

Reply:
done